components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1109,7 +1109,22 @@ where
 
             match res {
                 ApplyResult::None => {}
-                ApplyResult::Res(res) => results.push_back(res),
+                ApplyResult::Res(res) => {
+                    results.push_back(res);
+                    if self.wait_data {
+                        apply_ctx.committed_count -= committed_entries_drainer.len();
+                        let mut pending_entries =
+                            Vec::with_capacity(committed_entries_drainer.len());
+                        pending_entries.extend(committed_entries_drainer);
```

Comment:
The snapshot would be older than the current commit index, only applying these pending entries is not enough. How about skipping all entries when `wait_data = true` until receiving a message to notify the snapshot is applied. The msg should include entries ranging from (snapshot_index, commit_index)/

Reply:
And please add a test case to cover "The snapshot would be older than the current commit index"

Reply:
If the leader always send the latest snapshot(snapshot after switch witness), I think "The snapshot would be older than the current commit index" won't happen.